### ISSUE TITLE
Modifying `opslevel_group` without changing `parent` causes this value to be reset

### ISSUE THREAD

**Issue by matthewhughes-uw** [OpsLevel/terraform-provider-opslevel#84]
# Steps to reproduce

Create two groups like:

```hcl
resource "opslevel_group" "parent_group" {
  name = "Parent Group"
}

resource "opslevel_group" "child_group" {
  name = "Child Group"
  parent = opslevel_group.parent_group.alias
}
```

As expected, two groups are created and the "Parent Group" is the parent of "Child Group"

```console
$ terraform state show opslevel_group.child_group
# opslevel_group.child_group:
resource "opslevel_group" "child_group" {
    alias  = "child_group"
    id     = "Z2lkOi8vb3BzbGV2ZWwvTmFtZXNwYWNlczo6R3JvdXAvMjc3MQ"
    name   = "Child Group"
    parent = "parent_group"
}
```

Now update some value of the child group that isn't the parent, like:

```diff
resource "opslevel_group" "child_group" {
-  name = "Child Group"
+  name = "Child Group edit"
  parent = opslevel_group.parent_group.alias
}
```

and the plan is expected:

```console
$ terraform plan
<--- SNIP --->
Terraform will perform the following actions:

  # opslevel_group.child_group will be updated in-place
  ~ resource "opslevel_group" "child_group" {
        id      = "Z2lkOi8vb3BzbGV2ZWwvTmFtZXNwYWNlczo6R3JvdXAvMjc3MQ"
      ~ name    = "Child Group" -> "Child Group edit"
        # (3 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
```

However, if we apply and inspect again the `parent` has been removed (or set to `nil`) on the child:

```console
terraform state show opslevel_group.child_group
# opslevel_group.child_group:
resource "opslevel_group" "child_group" {
    alias        = "child_group_edit"
    id           = "Z2lkOi8vb3BzbGV2ZWwvTmFtZXNwYWNlczo6R3JvdXAvMjc3MQ"
    last_updated = "Wednesday, 03-May-23 09:23:07 UTC"
    members      = []
    name         = "Child Group edit"
}
```

This is reflected in the UI where this group is no longer a child of the parent group

# Version info

```console
$ terraform version 
Terraform v1.4.5
on linux_amd64
+ provider registry.terraform.io/opslevel/opslevel v0.8.2
```

**Comment by rocktavious** [comment]
@matthewhughes-uw - I'm going to try and get this fixed before the next release (next week - May 19th)

**Comment by rocktavious** [comment]
Fixed by https://github.com/OpsLevel/terraform-provider-opslevel/commit/5600787f73de9af4f4096bf37df410d666e74e5d - will be in the next release

**Comment by matthewhughes-uw** [comment]
> Fixed by [5600787](https://github.com/OpsLevel/terraform-provider-opslevel/commit/5600787f73de9af4f4096bf37df410d666e74e5d) - will be in the next release

thanks!
